jib-core/src/main/java/com/google/cloud/tools/jib/docker/CliDockerClient.java

```diff
@@ -103,30 +105,27 @@ public List<DescriptorDigest> getDiffIds() throws DigestException {
   public static final Path DEFAULT_DOCKER_CLIENT = Paths.get("docker");
 
   /**
-   * Checks if Docker is installed on the user's system and accessible by running the default {@code
-   * docker} command.
+   * Checks if Docker is installed on the user's system by running the `docker` command.
    *
    * @return {@code true} if Docker is installed on the user's system and accessible
    */
   public static boolean isDefaultDockerInstalled() {
-    return isDockerInstalled(DEFAULT_DOCKER_CLIENT);
+    try {
+      new ProcessBuilder(DEFAULT_DOCKER_CLIENT.toString()).start();
```

Comment:
Checking for existence of a file for the situation when the docker is installed on PATH has been a little tricky since there can be several places where docker could be install on a system (this also differs from OS to OS). For checking whether the path passed through `jib.dockerClient.executable` is installed correctly, however, we no longer run executable itself.